File: CppExample/example02/route_api_client.cpp
#include "route_api_client.h"
#include <iostream>

using namespace std;

string RouteAPIClient::getServiceName() {
    return "Route";
}

int RouteAPIClient::connect() {
    cout << "Connecting to " << this->getServiceName() << "..." << endl;
    cout << "Connected!" << endl;
    return 0;
};

int RouteAPIClient::get(std::string url) {
    cout << "Sending HTTP-GET request to " << this->getServiceName() << ": " << url << endl;
    return 200;
}

int RouteAPIClient::disconnect() {
    cout << "Disconnecting from " << this->getServiceName() << "..." << endl;
    cout << "Disconnected!" << endl;
    return 0;
}

File: JavaExample/src/main/java/hr/intis/cloud/examples/api/RouteAPIClient.java
package hr.intis.cloud.examples.api;

public class RouteAPIClient implements BasicAPIClient {

    public RouteAPIClient() {
    }

    @Override
    public String getServiceName() {
        return "Route";
    }

    @Override
    public int connect() {
        System.out.printf("Connecting to %s...\n", getServiceName());
        System.out.println("Connected!");
        return 0;
    }

    @Override
    public int get(String url) {
        System.out.printf("Sending HTTP-GET request to %s: %s\n", getServiceName(), url);
        return 200;
    }

    @Override
    public int disconnect() {
        System.out.printf("Disconnecting from %s...\n", getServiceName());
        System.out.println("Disconnected!");
        return 0;
    }
}
